packages/block-library/src/calendar/index.php

```diff
@@ -13,10 +13,14 @@
  * @return string Returns the block content.
  */
 function render_block_core_calendar( $attributes ) {
-	global $monthnum, $year, $wpdb;
+	global $monthnum, $year;
 
-	$has_posts = $wpdb->get_var( "SELECT 1 as test FROM $wpdb->posts WHERE post_type = 'post' AND post_status = 'publish' LIMIT 1" );
-	if ( ! $has_posts ) {
+	$has_published_posts = get_option( 'gutenberg_calendar_block_has_published_posts', null );
```

Comment:
Nice work! I think I would prefer to see this logic inside of the `block_core_calendar_update_has_published_posts` function itself. That way you don't need to know any implementation logic, just `if ( ! block_core_calendar_update_has_published_posts() ) { return; }`